src/api/src/main.py

```diff
@@ -69,60 +110,72 @@ async def handle_download(
+    with start_action(
+        action_type="download",
+        download_id=download_id,
+        download_images=download_images,
+        mode=mode,
+    ):
+        if username and not password or password and not username:
+            logger.error(
+                "Username with no Password or Password with no Username provided."
+            )
             return HTMLResponse(
-                status_code=403,
-                content='Incorrect Username and/or Password. Support is available on the <a href="https://discord.gg/P9RHC4KCwd" target="_blank">Discord</a>',
+                status_code=422,
+                content='Include both the username <u>and</u> password, or neither. Support is available on the <a href="https://discord.gg/P9RHC4KCwd" target="_blank">Discord</a>',
             )
-    else:
-        cookies = None
-
-    match mode:
-        case DownloadMode.story:
-            story_id = download_id
-        case DownloadMode.part:
-            story_id = await fetch_story_id(download_id, cookies)
-
-    book = epub.EpubBook()
-
-    metadata = await retrieve_story(story_id, cookies)
-    set_metadata(book, metadata)
-
-    await set_cover(book, metadata, cookies=cookies)
 
-    async for title in add_chapters(
-        book, metadata, download_images=download_images, cookies=cookies
-    ):
-        ...
-
-    # Book is compiled
-    temp_file = tempfile.NamedTemporaryFile(
-        suffix=".epub", delete=True
-    )  # Thanks https://stackoverflow.com/a/75398222
-
-    # create epub file
-    epub.write_epub(temp_file, book, {})
-
-    temp_file.file.seek(0)
-    book_data = temp_file.file.read()
-
-    return StreamingResponse(
-        BytesIO(book_data),
-        media_type="application/epub+zip",
-        headers={
-            "Content-Disposition": f'attachment; filename="{slugify(metadata["title"])}_{story_id}_{"images" if download_images else ""}.epub"'  # Thanks https://stackoverflow.com/a/72729058
-        },
-    )
+        if username and password:
+            # username and password are URL-Encoded by the frontend. FastAPI automatically decodes them.
+            try:
+                cookies = await wp_get_cookies(username=username, password=password)
+            except ValueError:
+                logger.error("Invalid username or password.")
+                return HTMLResponse(
+                    status_code=403,
+                    content='Incorrect Username and/or Password. Support is available on the <a href="https://discord.gg/P9RHC4KCwd" target="_blank">Discord</a>',
+                )
+        else:
+            cookies = None
+
+        match mode:
+            case DownloadMode.story:
+                story_id = download_id
+                metadata = await retrieve_story(story_id, cookies)
+            case DownloadMode.part:
+                story_id, metadata = await fetch_story_id(download_id, cookies)
+
+        logger.error(f"Retrieved story id ({story_id=})")
```

Comment:
Shouldn't this be `logger.info` instead of `logger.error`?

Reply:
Oh, that's right